cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/DFGPass.kt

```diff
@@ -466,7 +468,21 @@ class DFGPass(ctx: TranslationContext) : ComponentPass(ctx) {
      * `x[i]`.
      */
     protected fun handleSubscriptExpression(node: SubscriptExpression) {
-        node.prevDFGEdges += node.arrayExpression
+        if (node.access == AccessValues.WRITE) {
+                node.nextDFGEdges
+            } else {
+                node.prevDFGEdges
+            }
+            .add(node.arrayExpression) {
+                granularity =
+                    if ((node.subscriptExpression as? Literal<*>)?.value is Number) {
```

Comment:
You should pay 5 € into our deposit for bad code for this!

Please use something like
```kotlin
                val literalValue = (node.subscriptExpression as? Literal<*>)?.value
                granularity =
                    when (literalValue) {
                        is Number -> indexed(literalValue)
                        is String -> indexed(literalValue)
                        else -> partial(node.subscriptExpression)
                    }
```